EIPS/eip-7160.md

```diff
@@ -37,51 +37,63 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
-
-  /// @notice Get all token uris associated with a particular token
-  /// @dev If a token uri is pinned, the index returned should be the index in the string array
-  /// @param tokenId The identifier for the nft
-  /// @return index An unisgned integer that specifies which uri is pinned for a token (or the default uri if unpinned)
-  /// @return uris A string array of all uris associated with a token
-  function tokenURIs(uint256 tokenId) external view returns (uint256 index, string[] memory uris);
-
-  /// @notice Pin a specific token uri for a particular token
-  /// @param tokenId The identifier of the nft
-  /// @param index The index in the string array returned from the `tokenURIs` function that should be pinned for the token
-  function pinTokenURI(uint256 tokenId, uint256 index) external;
-
-  /// @notice Unpin metadata for a particular token
-  /// @dev This should reset the token to the default uri
-  /// @param tokenId The identifier of the nft
-  function unpinTokenURI(uint256 tokenId) external;
-
-  /// @notice Check on-chain if a token id has a pinned uri or not
-  /// @dev Useful for on-chain mechanics
-  /// @param tokenId The identifier of the nft
-  /// @return pinned A bool specifying if a token has metadata pinned or not
-  function hasPinnedTokenURI(uint256 tokenId) external view returns (bool pinned);
+/// @dev The ERC-165 identifier for this interface is 0x06e1bc5b.
+interface IERC7160 {
+    /// @dev This event emits when the token uris for a token have been updated
+    /// @dev This event MUST be emitted whenever a token uri is added, modified, or removed.
+    event TokenUrisUpdated(uint256 indexed tokenId);
+
+    /// @dev This event emits when a token uri is pinned and is
+    ///  useful for indexing purposes.
+    event TokenUriPinned(uint256 indexed tokenId, uint256 indexed index);
+
+    /// @dev This event emits when a token uri is unpinned and is
+    ///  useful for indexing purposes.
+    event TokenUriUnpinned(uint256 indexed tokenId);
+
+    /// @notice Get all token uris associated with a particular token
+    /// @dev If a token uri is pinned, the index returned SHOULD be the index in the string array
+    /// @dev This call MUST revert if the token does not exist
+    /// @param tokenId The identifier for the nft
+    /// @return index An unisgned integer that specifies which uri is pinned for a token (or the default uri if unpinned)
+    /// @return uris A string array of all uris associated with a token
+    /// @return pinned A boolean showing if the token has pinned metadata or not
+    function tokenURIs(uint256 tokenId) external view returns (uint256 index, string[] memory uris, bool pinned);
+
+    /// @notice Pin a specific token uri for a particular token
+    /// @dev This call MUST revert if the token does not exist
+    /// @dev This call MUST emit a `TokenUriPinned` event
+    /// @dev This call MAY emit a `MetadataUpdate` event from ERC-4096
```

Comment:
I feel like the semantics of TokenUrisUpdated and MetadataUpdate are the same and this will lead to confusion. We should pick one or the other